code/game/objects/items/weapons/cigs.dm

```diff
@@ -103,6 +90,8 @@ LIGHTERS ARE IN LIGHTERS.DM
 				explosion(user.loc, -1, 0, 2, 3, 0, flame_range = 2)
 			F.charges--
 
+	else if(is_hot(W) > 600)
```

Comment:
Save this for a different PR. `is_hot` is a terrible proc that violates OOP; likewise, the cig lighting needs to be made more OOP, as well.